packages/@lwc/engine-core/src/framework/fragment-cache.ts

```diff
@@ -0,0 +1,35 @@
+import { isUndefined } from '@lwc/shared';
+
+export const enum FragmentCacheKey {
+    HAS_SCOPED_STYLE = 1,
+    SHADOW_MODE_SYNTHETIC = 2,
+}
+
+// Mapping of cacheKeys to token array (assumed to come from a tagged template literal) to an Element
+const fragmentCache: Map<number, WeakMap<string[], Element>> = new Map();
```

Comment:
I considered just creating 4 `WeakMaps` but it didn't seem to offer a perf improvement.